lib/Pool.js

```diff
@@ -109,8 +109,11 @@ Pool.prototype.releaseConnection = function releaseConnection(connection) {
     }
 
     if (connection._purge) {
-      // purge connection from pool
-      this._purgeConnection(connection);
+      // restore the switched user
+      this.restoreUser(connection, function (err) {
+        if (err) return connection.destroy();
```

Comment:
I'm honestly not sure now.  Could have been an arbitary choice.  I've changed to `_purgeConnection` in 590256b and tests pass locally.